toonz/sources/toonz/ocaio.cpp

```diff
@@ -534,4 +535,379 @@ void ExportOCACommand::execute() {
+    // Ok you can set white level in the capture setting image ajust,
+    // and check White As Transparent in the raster level settings,
+    // The level premult option doesn't make onion skin work, if white level is too low...
+    // but that's destructive.
+    // And the level premult option doesn't make onion skin work, if white level is too low...
+    // I would even add a realtime shader for chromakey and lumakey in the column header 
+    // (even though i agree that it would be painfull to have all necessary keying parameters exposed column header in the ui)
+    // - Frame tags
+    // NavigationTags (frame tags) should be supported by OCA
+    // NavigationTags *navTags = xsh->getNavigationTags();
+    // this could be a great base to implement an onion skin filter... (like TVPaint has)
+    // (ex : onion skin displays only 1 prev + 1 next key drawing, etc...)
+    // Unfortunately, OCA doesn't support that.
+
+    // recursion for sublayers : nothing special to do ?
+    // see CHILD_XSHLEVEL ?
+    for (auto layer : jsonLayer["childLayers"].toArray()) {
+      // I guess we dont need to deal with jsonLayer["passThrough"] ?
+      // when passThrough is false, the group content *must* be merged in rendering, 
+      // else the group is only used as a way to group the layers in the UI of the application
+      importOcaLayer(layer.toObject());
+    }
+
+    TApp::instance()->getCurrentLevel()->setLevel(
+        level->getSimpleLevel());  // selects the last created level
+    TApp::instance()->getCurrentLevel()->notifyLevelChange();
+    TApp::instance()->getCurrentScene()->notifyCastChange();
+
+  } else {
+    DVGui::warning(QObject::tr("importOcaLayer skipping unimplemented layer type : ") +
+        jsonLayer["type"].toString() + " : " + jsonLayer["name"].toString()
+    );
+  }
+}
+
+/// <summary>
+/// creates frame from OCA frame
+/// see OCA frame specs
+/// https://github.com/RxLaboratory/OCA/blob/master/src-docs/docs/specs/frame.md
+/// UNUSED properties :
+/// - meta: metadata...
+/// - opacity: per frame opacity isn't used.
+/// - position: transforms handled in a separate file ? (keyframe animation transfert for camera, pegs, etc...) 
+/// And what about scale/rotation ?
+/// - name: could be used to create empty frames (named "_blank")
+/// </summary>
+/// <param name="jsonLayer"></param>
+
+void OCAIo::OCAInputData::importOcaFrame(const QJsonObject &jsonFrame,
+                                         TXshSimpleLevel *sl) {
+  // see LoadLevelPopup, TImageP TImageReader::load(), ResourceImportDialog
+  TFrameId fid(jsonFrame["frameNumber"].toInt());
+  // sl->setFrame(fid, TImageP());
+  auto path       = TFilePath(m_parentDir.getQString() + "/" +
+                        jsonFrame["fileName"].toString());
+  bool fileExists = TFileStatus(path).doesExist();
+  TImageP img;
+  if (fileExists) {
+    auto imageReader = TImageReaderP(path);
+    img = imageReader->load();
```

Comment:
May need to put this in a try/catch block since it's possible the load may fail. Additionally should verify `img` != 0 in the next line.

In my quick test, I OCA exported my test scene from a different T2D instance and then tried to Import it into this PR artifact and crashed on the next line because it failed to load one of my images.

Reply:
I believe, the reason why `img` is 0 here is because my level had blank frames before the 1st drawing.  The OCA Export created an entry for the blank frame but the Import doesn't handle empty frames.

```    "layers": [
        {
            "animated": true,
            "blendingMode": "normal",
            "childLayers": [
            ],
            "fileType": "png",
            "frames": [
                {
                    "duration": 1,
                    "fileName": "",
                    "frameNumber": 1,
                    "height": 0,
                    "name": "_blank",
                    "opacity": 1,
                    "position": [
                        960,
                        540
                    ],
                    "width": 0
                },
```

Reply:
@manongjohn 
That's strange, doesExists (QFileInfo.exists()) should have skip those empty frames...
Maybe we should check QFileInfo.exists() https://stackoverflow.com/questions/23930754/qfileinfo-exists-and-isfile-error
or use 
```
#include <sstream>
#include <iomanip>
static bool fileExists(std::string fileName) {
  std::ifstream infile(fileName.c_str());
  return infile.good();
}
```
found in my previous pr [iocommand.cpp](https://github.com/tahoma2d/tahoma2d/pull/1359/files#diff-c22e50ed5b65b43cdb2bd77320dae1eeb8526a5ae38420eeb3e0a900d9c3fd51) line 2284
Anyway that should be fixed...

Reply:
i've just tested, and QFileInfo correctly returns false with an empty string argument.. It's strange that you had that issue with empty frames, in previous version, but well...